packages/window-size-tool/rules.json

```
@@ -1,52 +1,57 @@
 [
   {
     "width": "< 1200",
-    "terms": ["//CLUSTER/@NUM_COLS > 4", "count(//LIST/*) > 6"],
+    "terms": ["//CLUSTER/@NUM_COLS > 4", "count(//LIST[count(FIELD | CONTAINER) > 6 ]) > 0"
+    ],
     "target": "xlg"
   },
   {
-    "width": ">= 768",
+    "width": "<= 768",
     "terms": [
       "//CLUSTER/@NUM_COLS >= 3",
       "count(//CLUSTER[@NUM_COLS >= 2]//LIST) > 0",
-      "count(//LIST/*) = 5 or count(//LIST/*) = 6"
+      "count(//LIST[count(FIELD | CONTAINER) > 4]) > 0 and count(//LIST[count(FIELD | CONTAINER) < 7]) > 0"
     ],
     "target": "lg"
   },
   {
     "width": "> 768",
     "terms": [
-      "//CLUSTER/@NUM_COLS = 2",
-      "count(//LIST) > 0 and count(//LIST/*) < 5"
+      "//CLUSTER/@NUM_COLS = 2 and count(//CLUSTER[@NUM_COLS >= 2]//LIST) = 0 and count(//CLUSTER[@NUM_COLS > 2]) = 0",
+      "count(//LIST[count(FIELD | CONTAINER) < 5]) > 0"
```

We also need to add protection here so the tool know when not to apply this rule.

```suggestion
      "count(//LIST[count(FIELD | CONTAINER) < 5]) > 0 and count(//LIST[count(FIELD | CONTAINER) > 4]) = 0"
```

We want to say only change a modal from large to medium if there are _no_ lists with more than 4 columns. If the page contains a list with 5  or 6 columns, it should be large.

I think we might also need to to add ` and count(//LIST[count(FIELD | CONTAINER) > 4]) = 0` to the other terms on line 20 also, because only one term needs to be true for the rule to apply then other terms are ignored.